packages/core/admin/ee/server/services/review-workflows/stages.js

```diff
@@ -61,9 +62,52 @@ module.exports = ({ strapi }) => {
       return strapi.db.transaction(async () => {
         const newStages = await this.createMany(created, { fields: ['id'] });
         const stagesIds = stages.map((stage) => stage.id ?? newStages.shift().id);
+        const defaultWorkflow = await getDefaultWorkflow({ strapi });
 
         await mapAsync(updated, (stage) => this.update(stage.id, stage));
-        await mapAsync(deleted, (stage) => this.delete(stage.id));
+
+        const entitiesToMove = [];
+        await mapAsync(deleted, async (stage) => {
+          // Find any entities related to this stage
+          const stageInfo = await this.findById(stage.id, {
+            workflowId: defaultWorkflow.id,
+            populate: ['related'],
+          });
+
+          if ((stageInfo?.related?.length ?? 0) === 0) {
```

Comment:
Can this be sufficient?
```suggestion
          if (!stageInfo?.related?.length) {
```